modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/impl/WFSContentDataAccess.java

```diff
@@ -184,6 +197,11 @@ public FeatureType getSchema(Name name) throws IOException {
      * @param cacheLocation the folder to use as the cache.
      */
     public void setCacheLocation(File cacheLocation) {
+        if (schemaParser != null && this.cacheLocation == null) {
```

Comment:
What if multiple stores are using the same cache location though? It might remove bits that are required, but worse, it could be deleting cached schemas while they are in use/being read.

I see one can configure the schema location from the store parameter SCHEMA_CACHE_LOCATION.

Cleaning up looks sensible in case the cache location was created with a random temp directory (I see it happening later in the diff) but probably not if it's externally configured.

@NielsCharlier opinions on this one? (I see you were involved in the initial development and worked on the schema location support too).

Reply:
The code for deleting is just for temporary folder, and that will be created by one WFSContentDataAccess, and deleted at dispose().